src/Kestrel.Tls/TlsStream.cs

```diff
@@ -181,9 +182,16 @@ public override Task WriteAsync(byte[] buffer, int offset, int count, Cancellati
                     {
                         var error = OpenSsl.SSL_get_error(_ssl, ret);
 
-                        if (error != 2)
+                        if (error == 1)
                         {
-                            throw new IOException($"TLS handshake failed: {nameof(OpenSsl.SSL_do_handshake)} error {error}.");
+                            // SSL error, get it from the OpenSSL error queue
+                            error = OpenSsl.ERR_get_error();
+                            if (error == 336130204)
```

Comment:
![http://www.reactiongifs.com/r/mye.gif](http://www.reactiongifs.com/r/mye.gif)

Reply:
Enjoy :). I'll put a constant in before merging.

Reply:
If this were anything except code that literally cannot be shipped I'd do something better. Also, it's just to clean up an ugly and hard to identify error message.